### DISCUSSION TITLE
How to set a spectrum of C+L band？

### DISCUSSION THREAD

**Discussion by hhingdihun** [Telecominfraproject/oopt-gnpy#508] [Q&A] [answered]
Hi Team,
I would like to know how to set the spectrum of C+L band, if the spectrum is created using create_input_spectral_information method, how to set the guard bandwidth?
For example: I want to set the spectrum as: from 186.1THz to 196.1Thz, 64 channels in each band, spectral grid of 75GHz, guard grid of 500GHz for C-band and L-band, how should I set it?
Thanks,
Daniel

**Comment by EstherLerouzic** [comment, marked as answer]
Dear @hhingdihun 

Thanks for using GNPy!

You may set the C+L spectrum by using the --spectrum option defining your two spectrum partitions. https://gnpy.readthedocs.io/en/master/json.html#equalization-choices:~:text=much%20as%20possible.-,SpectralInformation,-%C2%B6
for example stating a spectrum like this one:

`
{
      "spectrum":[
            {
            "f_min": 186.1e12,
            "f_max":190.9e12,
            "baud_rate": 32e9,
            "slot_width": 75e9,
            "delta_pdb": 0,
            "roll_off": 0.15,
            "tx_osnr": 40,
            "label": "mode_1"
            },
            {
            "f_min": 191.3e12,
            "f_max":196.1e12,
            "baud_rate": 32e9,
            "slot_width": 75e9,
            "roll_off": 0.15,
            "tx_osnr": 40,
            "label": "mode_2"
            }
      ]
}
`

However, please note that although the fiber modelling properly supports multiband case, with proper SRS interband power transfer, the current release does not support the modelling of C+L amps yet, so that the same Edfa is used for both bands. The C+L feature is under preparation and a first draft (incomplete and unstable) is available here: https://review.gerrithub.io/c/Telecominfraproject/oopt-gnpy/+/555555.

Hope this helps !
